5.4.0 changelog (#6571)
hashicorp/terraform-provider-google-beta: CHANGELOG.md

## 5.4.0 (Unreleased)

> NOTES:

Reviewer: ```suggestion
DEPRECATIONS:
* bigquery: deprecated `cloud_spanner.use_serverless_analytics` on `google_bigquery_connection`. Use `cloud_spanner.use_data_boost` instead. ([#6539](https://github.com/hashicorp/terraform-provider-google-beta/pull/6539))

NOTES:
```

* cloudrunv2: marked `location` field as required in resource `google_cloud_run_v2_service`. Any configuration without `location` specified will cause resource creation fail ([#6540](https://github.com/hashicorp/terraform-provider-google-beta/pull/6540))

FEATURES:
* **New Data Source:** `google_cloud_identity_group_lookup` ([#6530](https://github.com/hashicorp/terraform-provider-google-beta/pull/6530))
* **New Resource:** `google_network_connectivity_policy_based_route` ([#6552](https://github.com/hashicorp/terraform-provider-google-beta/pull/6552))
* **New Resource:** `google_pubsub_schema_iam_*` ([#6533](https://github.com/hashicorp/terraform-provider-google-beta/pull/6533))

IMPROVEMENTS:
* accesscontextmanager: added support for Access Context Manager's VPC Sub-segmentation feature ([#6553](https://github.com/hashicorp/terraform-provider-google-beta/pull/6553))
* apigee: added support for type in `google_apigee_environment` ([#6562](https://github.com/hashicorp/terraform-provider-google-beta/pull/6562))

> * bigquery: added data boost and parallelism input options to `google_bigquery_connection` ([#6539](https://github.com/hashicorp/terraform-provider-google-beta/pull/6539))

Reviewer: ```suggestion
* bigquery: added `cloud_spanner.database_role`, `cloud_spanner.use_data_boost`, and `cloud_spanner.max_parallelism` fields to `google_bigquery_connection` ([#6539](https://github.com/hashicorp/terraform-provider-google-beta/pull/6539))
```

* container: added `enable_confidential_storage` to `node_config` in `google_container_cluster` and `google_container_node_pool` ([#6531](https://github.com/hashicorp/terraform-provider-google-beta/pull/6531))
* container: added update support for `google_container_node_pool.node_config.taint` ([#6536](https://github.com/hashicorp/terraform-provider-google-beta/pull/6536))
* containerattached: added `admin_groups` field to `google_container_attached_cluster` resource ([#6537](https://github.com/hashicorp/terraform-provider-google-beta/pull/6537))
* dialogflowcx: added `advanced_settings` field to `google_dialogflow_cx_flow` resource ([#6543](https://github.com/hashicorp/terraform-provider-google-beta/pull/6543))
* dialogflowcx: added `advanced_settings` fields to `google_dialogflow_cx_page` resource ([#6543](https://github.com/hashicorp/terraform-provider-google-beta/pull/6543))
* dialogflowcx: added `advanced_settings`, `text_to_speech_settings`, `git_integration_settings` fields to `google_dialogflow_cx_agent` resource ([#6543](https://github.com/hashicorp/terraform-provider-google-beta/pull/6543))

> * redis: allow `replica_count` to be set to zero in the `google_redis_cluster` resource ([#6534](https://github.com/hashicorp/terraform-provider-google-beta/pull/6534))

Reviewer: This should be a bug rather than an enhancement.

> * sql: added `ssl_mode` field to `google_sql_database_instance` resource ([#6564](https://github.com/hashicorp/terraform-provider-google-beta/pull/6564))

Reviewer 2: ```suggestion
```
since this change is reverted

> * tpuv2: added more fields to `google_tpu_v2_vm` resource, including `network_config`, `scheduling_config`, `shielded_instance_config`, `service_account` and `data_disks` ([#6555](https://github.com/hashicorp/terraform-provider-google-beta/pull/6555))

Reviewer: could we maybe list all of the added fields instead of just a few of them?

> * tpuv2: added`accelerator_config` field to `google_tpu_v2_vm` resource ([#6559](https://github.com/hashicorp/terraform-provider-google-beta/pull/6559))

Reviewer 3: ```suggestion
* tpuv2: added `accelerator_config` field to `google_tpu_v2_vm` resource ([#6559](https://github.com/hashicorp/terraform-provider-google-beta/pull/6559))
```

BUG FIXES:
* bigquery: fixed a bug when updating a `google_bigquery_dataset` that contained an `iamMember` access rule added out of band with Terraform ([#6550](https://github.com/hashicorp/terraform-provider-google-beta/pull/6550))
* bigqueryreservation: fixed bug of incorrect resource recreation when `capacity_commitment_id` is unspecified in resource `google_bigquery_capacity_commitment` ([#6548](https://github.com/hashicorp/terraform-provider-google-beta/pull/6548))

> * cloudrunv2: set field `annotations` having all of annotations present on the resource in GCP in data source `google_cloud_run_v2_job` ([#6532](https://github.com/hashicorp/terraform-provider-google-beta/pull/6532))

Reviewer: Same change below.

```suggestion
* cloudrunv2: made `annotations` field on the `google_cloud_run_v2_job` data source include all annotations present on the resource in GCP ([#6532](https://github.com/hashicorp/terraform-provider-google-beta/pull/6532))
```

NOTES:
* provider: added `universe_domain` attribute as a provider attribute ([#6551](https://github.com/hashicorp/terraform-provider-google-beta/pull/6551))

BREAKING CHANGES:
* cloudrunv2: marked `location` field as required in resource `google_cloud_run_v2_job`. Any configuration without `location` specified will cause resource creation fail ([#6540](https://github.com/hashicorp/terraform-provider-google-beta/pull/6540))
* cloudrunv2: marked `location` field as required in resource `google_cloud_run_v2_service`. Any configuration without `location` specified will cause resource creation fail ([#6540](https://github.com/hashicorp/terraform-provider-google-beta/pull/6540))

FEATURES:
* **New Data Source:** `google_cloud_identity_group_lookup` ([#6530](https://github.com/hashicorp/terraform-provider-google-beta/pull/6530))
* **New Resource:** `google_network_connectivity_policy_based_route` ([#6552](https://github.com/hashicorp/terraform-provider-google-beta/pull/6552))
* **New Resource:** `google_pubsub_schema_iam_*` ([#6533](https://github.com/hashicorp/terraform-provider-google-beta/pull/6533))

IMPROVEMENTS:

> * accesscontextmanager: added support for Access Context Manager's VPC Sub-segmentation feature ([#6553](https://github.com/hashicorp/terraform-provider-google-beta/pull/6553))

Reviewer: I think it might make sense to specify the fields that were added rather than the feature?

> * apigee: added support for type in `google_apigee_environment` ([#6562](https://github.com/hashicorp/terraform-provider-google-beta/pull/6562))

Reviewer: ```suggestion
* apigee: added support for `type` in `google_apigee_environment` ([#6562](https://github.com/hashicorp/terraform-provider-google-beta/pull/6562))
```
